src/main/java/com/minecolonies/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -0,0 +1,509 @@
+        final AbstractBuilding building = (AbstractBuilding) buildingArray[random];
+
+        if (building instanceof BuildingGuardTower
+                || BlockPosUtil.getDistance2D(building.getLocation(), this.getOwnBuilding().getLocation()) > getPatrolDistance())
+        {
+            return this.getOwnBuilding().getLocation();
+        }
+
+        return building.getLocation();
+    }
+
+    /**
+     * Called when a guard killed an entity.
+     *
+     * @param killedEntity the entity being killed.
+     */
+    protected void onKilledEntity(final EntityLivingBase killedEntity)
+    {
+        final Colony colony = this.getOwnBuilding().getColony();
+        colony.incrementMobsKilled();
+        incrementActionsDone();
+    }
+
+    /**
+     * Getter for the patrol distance the guard currently has.
+     * @return the distance in whole numbers.
+     */
+    private int getPatrolDistance()
+    {
+        return this.getOwnBuilding().getBuildingLevel() * PATROL_DISTANCE;
+    }
+
+    /**
+     * Checks if the guard found items on the ground,
+     * if yes collect them, if not search for them.
+     *
+     * @return GUARD_GATHERING as long as gathering takes.
+     */
+    private AIState gathering()
+    {
+        if (items == null)
+        {
+            searchForItems();
+        }
+        if (!items.isEmpty())
+        {
+            gatherItems();
+            return getState();
+        }
+        items = null;
+        return GUARD_PATROL;
+    }
+
+    @Override
+    protected boolean neededForWorker(@Nullable final ItemStack stack)
+    {
+        return stack != null
+                && (stack.getItem() instanceof ItemArmor
+                || stack.getItem() instanceof ItemTool
+                || stack.getItem() instanceof ItemSword
+                || stack.getItem() instanceof ItemBow);
+    }
+
+    /**
+     * Search for all items around the Guard.
+     * and store them in the items list
+     */
+    private void searchForItems()
+    {
+        items = new ArrayList<>();
+
+        items = world.getEntitiesWithinAABB(EntityItem.class, worker.getEntityBoundingBox().expand(RANGE_HORIZONTAL_PICKUP, RANGE_VERTICAL_PICKUP, RANGE_HORIZONTAL_PICKUP))
+                .stream()
+                .filter(item -> item != null && !item.isDead)
+                .map(BlockPosUtil::fromEntity)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Collect one item by walking to it
+     */
+    private void gatherItems()
+    {
+        worker.setCanPickUpLoot(true);
+        if (worker.getNavigator().noPath())
+        {
+            BlockPos pos = getAndRemoveClosestItem();
+            worker.isWorkerAtSiteWithMove(pos, ITEM_PICKUP_RANGE);
+            return;
+        }
+        if (worker.getNavigator().getPath() == null)
+        {
+            setDelay(WAIT_WHILE_WALKING);
+            return;
+        }
+
+        int currentIndex = worker.getNavigator().getPath().getCurrentPathIndex();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'currentIndex' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)